### PR TITLE
JENKINS-44625# Fix for exit and cog wheel links with custom organization

### PR DESCRIPTION
**Pull request by vivek**
# Description

See [JENKINS-44625](https://issues.jenkins-ci.org/browse/JENKINS-44625)

cc: @alvarolobato 

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
there seems to be a :bug: which is adding in more '/' than is needed. 

This is picked up in a failing test: 
```
1) UrlUtils build classicConfigUrl should build the url for classic config:
     AssertionError: expected '/jenkins//job//foldey/job/nesty/job/woozle%20wozzle/job/mazzig/configure' to equal '/jenkins/job/foldey/job/nesty/job/woozle%20wozzle/job/mazzig/configure'
      at Context.<anonymous> (src/test/js/UrlUtils-spec.js:153:20)
```

the test: 
```
    describe('build classicConfigUrl', () => {

        const testData = {
            fullName : 'foldey/nesty/woozle%20wozzle/mazzig'
        };

        it('should build the url for classic config', () => {
            const url = buildClassicConfigUrl(testData);
            assert.equal(url, '/jenkins/job/foldey/job/nesty/job/woozle%20wozzle/job/mazzig/configure');
        });

    });
```

@vivek perhaps @cliffmeyers could take this one on this week as this looks purely front end?

**Comment by vivek** [comment]
@michaelneale Ah, of course, there was no need to prefix and suffix return of jobPrefixPath() as it always returns values with '/' prefix/suffix. This was small fix that I just made it. It should be all be good. I should have ran tests, was confident as it worked in manual testing as backend handled extra '/'. Should be good now.

@cliffmeyers PTAL anyways and feel free to rewrite or make changes as it seems fit.

**Comment by michaelneale** [comment]
seems to work well for me - perhaps @cliffmeyers can review/bless (assuming build passes) but I suspect this is ok (just remember when merging, need to do the publish dance with core-js still)

File: blueocean-core-js/src/js/utils/UrlUtils.js [hunk 1]
```diff
@@ -274,7 +283,7 @@ export function toClassicJobPage(pageUrl, isMultibranch = false) {
         // The next token is the "full" job name, URL encoded.
         const fullJobName = decodeURIComponent(pageUrlTokens.shift());
         const fullJobNameTokens = fullJobName.split('/');
-        const classicJobFullName = '/job/' + fullJobNameTokens.join('/job/');
+        const classicJobFullName = jobPrefixPath(AppConfig.getOrganizationName()) + fullJobNameTokens.join('/job/');
```
Comment (alvarolobato) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: This looks wrong to me, it should be that should be the path of the organization group.
https://github.com/jenkinsci/blueocean-plugin/blob/8d8d5234579167ebbc072afda09d554d002ec8b0/blueocean-rest/src/main/java/io/jenkins/blueocean/rest/factory/organization/AbstractOrganization.java#L21

Prepending the path of `organization.getGroup()` should be enough

Comment (vivek) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: @alvarolobato ack. Fix on the way.

File: blueocean-core-js/src/js/utils/UrlUtils.js [hunk 2]
```diff
@@ -1,4 +1,13 @@
 import AppConfig from '../config';
+
+function jobPrefixPath(organization) {
+    // custom organization
+    if (organization !== 'undefined' && organization !== 'jenkins') {
```
Comment (cliffmeyers) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: is the value of the string ever really "undefined"? That seems a little odd to me. Perhaps you could elaborate on this?

Comment (vivek) [inline blueocean-core-js/src/js/utils/UrlUtils.js]: This was to get unit test work, its changing a little, wait for my next commit.

**Comment by vivek** [comment]
@alvarolobato I have answered your concerns, PTAL.

**Comment by vivek** [comment]
@alvarolobato yes, thats the plan. Let me fix this failing test then should be ready to be merged.

**Comment by vivek** [comment]
Huh, it passes locally.

```
7 passing (54ms)
[INFO]   1 failing
[INFO] 
[INFO]   1) PipelineCard renders with correct props:
[INFO]      AssertionError: expected '2016-06-16T01:31:03.830-0400' to equal '2016-05-24T08:57:04.432-0400'
[INFO]       at Context.<anonymous> (src/test/js/components/PipelineCard-spec.jsx:65:16)
[INFO] 
[INFO] 
```

@cliffmeyers This seems flaky, for some reason test is picking up second favorite data and not the first. May be favorite order changed?  Anyways, this failure looks unrelated to this PR.

**Comment by vivek** [comment]
Let's see this build. I have pushed released version of blueocean-core-js.

**Comment by vivek** [comment]
Cherry picked to 1.1.0-release branch.

**Pull request merged**